onpremise/onpremise.go

```diff
@@ -319,10 +416,99 @@ func (p *Engine) validateAndAppendUrlParams() error {
 	return nil
 }
 
-func (p *Engine) hasDefaultDistributorParams() bool {
-	return len(p.licenceKey) > 0 && len(p.product) > 0
+func (e *Engine) hasDefaultDistributorParams() bool {
+	return len(e.licenceKey) > 0 && len(e.product) > 0
 }
 
-func (p *Engine) hasSomeDistributorParams() bool {
-	return len(p.licenceKey) > 0 || len(p.product) > 0
+func (e *Engine) hasSomeDistributorParams() bool {
+	return len(e.licenceKey) > 0 || len(e.product) > 0
+}
+
+func (e *Engine) handleFileExternallyChanged() {
+	e.Lock()
+	defer e.Unlock()
+	if e.isCreateTempDataCopyEnabled {
+		dirPath, filePath, err := e.copyToTempFile()
+		if err != nil {
+			e.logger.Printf("failed to copy data file: %v", err)
+		}
+		fullPath := filepath.Join(dirPath, filePath)
+		err = e.replaceManager(fullPath)
+		if err != nil {
+			e.logger.Printf("failed to replace manager: %v", err)
+		}
+
+		oldFullPath := filepath.Join(e.tempDataDir, e.tempDataFile)
+		err = os.Remove(oldFullPath)
+		if err != nil {
+			e.logger.Printf("failed to remove old temp data file: %v", err)
+		}
+	} else {
+		err := e.replaceManager(e.dataFile)
+		if err != nil {
+			e.logger.Printf("failed to replace manager: %v", err)
+		}
+	}
+}
+
+func (e *Engine) replaceManager(filePath string) error {
+	if !e.isCreateTempDataCopyEnabled {
```

Comment:
i renamed function replaceOrInitManager, and added comments to explain what it does

// this function will be called when the engine is started or the is new file available
// it will create and initialize a new manager from the new file if it does not exist
// if the manager exists, it will create a new manager from the new file and replace the existing manager thus freeing memory of the old manager